### DISCUSSION TITLE
OrderLine question

### DISCUSSION THREAD

**Discussion by dwightkelly** [vendurehq/vendure#683] [Q&A] [answered]
We are attempting to implement a workflow where when a product, that doesn't have any variants, is added to the cart more than once it gets added as a unique OrderLine and doesn't just increment quantity.

**Comment by michaelbromley** [comment, marked as answer]
Hi,
Can you explain a bit more about this use-case so I have a better understanding of why you need it to work like this? Then I might be able to suggest something that is workable.

**Reply by dwightkelly** [reply]
Customer selects product1 and modifies the artwork using our custom widget. This gets added to cart as an OrderLine with customfield containing artwork identifier. They then add product1 again but with different artwork. We don't want the original OrderLine to be updated to quantity 2. Rather, we want a 2nd OrderLine with the customfield containing the artwork for 2nd product1.

Predefined variants won't work as the artwork is supplied by customer at time of purchase.

**Reply by michaelbromley** [reply]
Thanks for the clarification. In this case Vendure should already work as you intend. When adding an item to the order, both the productVariantId _and_ the customFields are checked to decide whether a new OrderLine should be created.

**Reply by dwightkelly** [reply]
Thank you
